CoreFoundation/NumberDate.subproj/CFTimeZone.c

```diff
@@ -1058,6 +1175,53 @@ Boolean _CFTimeZoneInitInternal(CFTimeZoneRef timezone, CFStringRef name, CFData
     return success;
 }
 
+CFDataRef _CFTimeZoneDataCreate(CFURLRef baseURL, CFStringRef tzName) {
+#if TARGET_OS_ANDROID
+    CFDataRef data = NULL;
+    char *buffer = NULL;
+    const char *tzNameCstr = CFStringGetCStringPtr(tzName, kCFStringEncodingASCII);
+    if (!tzNameCstr) {
+        CFIndex maxSize = CFStringGetMaximumSizeForEncoding(CFStringGetLength(tzName), kCFStringEncodingASCII) + 2;
+        if (maxSize == kCFNotFound) {
+            return NULL;
+        }
+        buffer = malloc(maxSize);
+        if (!buffer) {
+            return NULL;
+        }
+        if (CFStringGetCString(tzName, buffer, maxSize, kCFStringEncodingASCII)) {
+            tzNameCstr = buffer;
+        }
+    }
+    if (!tzNameCstr) {
+        return NULL;
```

Comment:
Does this need a `if (buffer) { free(buffer); }` before the return because the `CFStringGetCString()`  could fail and `tzNameCStr` could still be `NULL` ?

Reply:
No, it doesn't need it.  `free(NULL)` is well specified.

Reply:
I mean wouldnt the `buffer = malloc(maxSize)` be leaked if  `CFStringGetCString` returned 0 as `tzNameCstr` would still be `NULL` and will return on line 1197 ?

Reply:
I never remember that `free(NULL)` is actually valid. I will correct an instance below.

I will add the `free(buffer)` that simons proposes. I don’t know why I thought that I will be fine here without it, but `CFStringGetCString` might still fail because of the encoding (but `__nameStringOK` should take care of most of those cases anyway).